src/Engine/Component/Script.cpp

```diff
@@ -25,5 +26,29 @@ Json::Value Script::Save() const {
     if (scriptFile != nullptr)
         component["scriptName"] = scriptFile->path + scriptFile->name;
     
+    for (auto &name_map : propertyMap) {
+
+        std::string name = name_map.first.c_str();
```

Comment:
Here we construct a string by getting the C-string of another instead of direct assignment (which ought to be more efficient). Even so, the string is never actually modified, so it would be even better to just hold a reference to begin with.

    const std::string& name = name_map.first;